tests/test_orient_sync.py

```diff
@@ -69,4 +69,4 @@ def test03EstRotations(self):
     def test04EstShifts(self):
         results = np.load(os.path.join(DATA_DIR, 'orient_est_shifts.npy'),
                           allow_pickle=True)
-        self.assertTrue(np.allclose(results, self.est_shifts, atol=1.0e-4))
```

Comment:
before we guess and change the code too much more to try and cover up whatever is happening here, lets figure out if there is actually bugs in the code.  I see all kinds of diagnostics for lsqr.  You might try dumping those diagnostics, shift_equations, shift_b in a different branch and seeing what the CI reports when you push that. Lets not debug it in your PR.

Reply:
also, restarting the various CI repeatedly until they pass is not going to be acceptable.

Reply:
I just did this for a quick test of the failing. Related commits have be removed. I will create anther branch for more tests.

Reply:
I created a separate PR #214  to do the failing test of estimate_shifts. However, there is no any failure when runing tests at different time and commits. Probably this is not related to the code.

Reply:
I think this was sorted out. Resolving